.github/workflows/deploy-github-page.yml

```diff
@@ -0,0 +1,106 @@
+name: Build and Deploy
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+    - main
+    
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+    
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - name: Github-page
+            os: ubuntu-24.04
+            emsdk_ver: "3.1.45"
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+        
+      - name: install mamba
+        uses: mamba-org/setup-micromamba@main
+        with:
+          environment-file: environment-wasm-build.yml
+          init-shell: bash
+          environment-name: xeus-r-wasm-build
+
+      - name: Setup default Build Type on *nux
+        if: ${{ runner.os != 'windows' }}
+        run: |
+          echo "ncpus=$(nproc --all)" >> $GITHUB_ENV
+          
+      - name: Setup emsdk
+        shell: bash -l {0}
+        run: |
+          cd $HOME
+          git clone https://github.com/emscripten-core/emsdk.git
+          cd emsdk
+          ./emsdk install ${{ matrix.emsdk_ver }}    
+
+      - name: Build xeus-r
+        shell: bash -l {0}
+        run: |
+          $HOME/emsdk/emsdk activate ${{matrix.emsdk_ver}}
+          source $HOME/emsdk/emsdk_env.sh
+          micromamba create -f environment-wasm-host.yml --platform=emscripten-wasm32
+          
+          mkdir build
+          pushd build
+
+          export PREFIX=$MAMBA_ROOT_PREFIX/envs/xeus-r-wasm-host 
+          echo "PREFIX=$PREFIX" >> $GITHUB_ENV
+          export CMAKE_PREFIX_PATH=$PREFIX
+          export CMAKE_SYSTEM_PREFIX_PATH=$PREFIX
+
+          emcmake cmake \
+            -DCMAKE_BUILD_TYPE=Release                        \
+            -DCMAKE_PREFIX_PATH=$PREFIX                       \
+            -DCMAKE_INSTALL_PREFIX=$PREFIX                    \
+            -DXEUS_R_EMSCRIPTEN_WASM_BUILD=ON                 \
+            -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ON            \
+            ..
+          emmake make -j ${{ env.ncpus }} install
+
+      - name: Jupyter Lite integration 
+        shell: bash -l {0}
+        run: |
+          micromamba create -n xeus-lite-host jupyterlite-core
+          micromamba activate xeus-lite-host
+          python -m pip install jupyterlite-xeus
+          jupyter lite build \
+              --XeusAddon.prefix=${{ env.PREFIX }} \
+              --XeusAddon.mount=share/jupyter/kernels/xr/resources \
+              --output-dir dist
+
+          mkdir -p dist/files
+          mv notebooks dist/files
```

Comment:
The notebook example won't work yet because it has some plotting stuff. I think we should add a new notebook only for JupyterLite with the examples that are currently working if we are going to share this with others.